srcs/ConfigSetup.cpp

```diff
@@ -1,56 +1,62 @@
 #include "ConfigSetup.hpp"
+#include <ostream>
 
-ConfigSetup::ConfigSetup(int argc, char *argv[]) {
-    if (argc < 2) {
-        config = config_from_default();
-    } else {
-        config = config_from_argument(argv[1]);
-    }
+ConfigSetup::ConfigSetup(int argc, char *argv[]): argc(argc), argv(argv), config(NULL) {
 }
 
 ConfigSetup::~ConfigSetup(){        
-    if (config != NULL)
-        delete config;
-    }
+    // if (config != NULL)
+    //     delete config;
+}
 
 Config* ConfigSetup::get_config() {
+    Config* config = NULL;
```

Comment:
porque isto e nao utilizar so this->config? Se nao nem se quer se esta a utilizar o config da class

Reply:
nas funcoes a seguir podia-se logo fazer return de, por exemplo get_config_from_file em vez de estar sempre a igualar a uma Config class